client/src/cordova/dev.action.mjs

```diff
@@ -0,0 +1,142 @@
+
+const OUTPUT_PATH = 'output/build/client/macos';
+const OUTLINE_APP_PATH = 'Debug/Outline.app';
+const OUTLINE_APP_WWW_PATH = 'Contents/Resources/www';
+const RELOAD_SERVER_PORT = 35729;
+
+const getUIHash = async () => {
+  const hashResult = await hashElement(
+    path.join(getRootDir(), 'client/src/www'),
+    {
+      files: {include: ['**/*.ts', '**/*.html', '**/*.css', '**/*.js']},
+    }
+  );
+
+  return hashResult.hash;
+};
+
+/**
+ * @description Runs the cordova app in development mode.
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...givenParameters) {
+  const {
+    _: [platform = 'macos'],
+    buildMode = 'release',
+    sentryDsn = 'https://public@sentry.example.com/1',
+    versionName = '0.0.0-dev',
+  } = minimist(givenParameters);
+
+  if (platform !== 'macos') {
+    throw new Error('This action currently only works for the MacOS platform.');
+  }
+
+  if (buildMode !== 'release') {
+    throw new Error('MacOS only renders properly in the release build mode.');
+  }
+
+  if (os.platform() !== 'darwin') {
+    throw new Error('You must be on MacOS to develop for MacOS.');
+  }
+
+  const parameters = [
+    'macos',
+    '--buildMode=release',
+    `--sentryDsn=${sentryDsn}`,
+    `--versionName=${versionName}`,
+  ];
+
+  await runAction('client/src/www/build', ...parameters);
+  await runAction('client/go/build', ...parameters);
+  await runAction('client/src/cordova/setup', ...parameters);
+
+  await makeReplacements([
+    {
+      files: path.join(
+        getRootDir(),
+        'client/platforms/osx/**/index_cordova.html'
+      ),
+      from: '<app-root></app-root>',
+      to: `<app-root></app-root>
+
+    <script>
+      try {
+        const reloadSocket = new WebSocket("ws://localhost:35729");
+
+        reloadSocket.onopen = () => console.log("LiveReload connected~");
+        reloadSocket.onmessage = ({ data }) => data === "reload" && location.reload();
+      } catch (e) {
+        // nevermind
+      }
+    </script>`,
+    },
+  ]);
+
+  await spawnStream(
+    'xcodebuild',
+    '-scheme',
+    'Outline',
+    '-workspace',
+    path.join(getRootDir(), 'client/src/cordova/apple/macos.xcworkspace'),
+    `SYMROOT=${path.join(getRootDir(), OUTPUT_PATH)}`
+  );
+
+  await spawnStream(
+    'open',
+    path.join(getRootDir(), OUTPUT_PATH, OUTLINE_APP_PATH)
+  );
+
+  let previousUIHashResult = await getUIHash();
+
+  console.log(`Starting reload server @ port ${RELOAD_SERVER_PORT}...`);
+  createReloadServer(async () => {
+    const currentUIHashResult = await getUIHash();
+
+    if (previousUIHashResult === currentUIHashResult) {
+      return false;
+    }
+
+    previousUIHashResult = currentUIHashResult;
+
+    await runAction('client/src/www/build', ...parameters);
+
+    await fs.copy(
+      path.join(getRootDir(), 'client/www'),
+      path.join(OUTPUT_PATH, OUTLINE_APP_PATH, OUTLINE_APP_WWW_PATH)
+    );
+
+    return true;
+  }).listen(RELOAD_SERVER_PORT);
```

Comment:
I believe you are listening on the "any" address here. This is a security issue. Please change this to localhost.

Also, use port zero, so that the OS assign a port for you. Then read the port (really, the full address) to pass it to the websocket url.